test/test/alloc/AllocTests.java

```diff
@@ -55,4 +60,16 @@ public void humongous(TestProcess p) throws Exception {
         assert out.contains("java/io/ByteArrayOutputStream.toByteArray;");
         assert out.contains("G1CollectedHeap::humongous_obj_allocate");
     }
+
+    @Test(mainClass = MapReaderOpt.class, jvmVer = {11, Integer.MAX_VALUE})
+    public void objectSamplerWtihDifferentAsprofs(TestProcess p) throws Exception {
+        Output out = p.profile("-e alloc -d 3 -o collapsed");
+        // _[k] suffix in collapsed output corresponds to jdk.ObjectAllocationOutsideTLAB, which means alloc tracer is being used
+        assert !out.contains("_\\[k\\]"); // we are using alloc tracer instead of object sampler, should definitely not happen on first profiling call
+        File asprofCopy = File.createTempFile("libAsyncProfiler", p.currentOs().getLibExt());
```

Comment:
The name is lib**a**syncProfiler.so (small `a`)
Or you can just extract a file name from `p.profilerLibPath()`